### PR TITLE
fix: Check planfile and put it in the end of an arguments list

### LINKED ISSUE #1271: terragrunt apply-all specific plan fails
This is not a new issue, and I can't believe it wasn't fixed so far.
Running `terragrunt apply-all` blindly is the perfect recipe for disaster.
In order to make this more acceptable for automation, we should at least allow for a reviewed plan to be applied.
The issue resides with  forced silent arguments `-input=false` and `-auto-approve`.
These arguments are added after any `extra_arguments`, which generates invalid terraform command
At least allow us to override your forced arguments.

```sh
# Generated command
terraform apply reviewed-plan.tfplan -input=false -auto-approve

...
Too many command line arguments. Configuration path expected.
```

### PR DESCRIPTION
**Pull request by PertsevRoman**
This issue should fix #1271

This MR fixes args order when planfile is specified in extra_args for apply

### CODE REVIEW

**Review by alexey-eshmetov — DISMISSED** [review]
👍

File: options/options.go [hunk 1]
```diff
@@ -302,8 +302,40 @@ func (terragruntOptions *TerragruntOptions) Clone(terragruntConfigPath string) *
...
+// Check if argument is planfile TODO check file format
+func checkIfPlanfile(arg string) bool {
+	stat, err := os.Stat(arg)
+
+	if err != nil {
+		return false
+	}
+
+	return !stat.IsDir()
+}
```
Comment (yorinasub17) [inline options/options.go:314]: You can use [util.IsFile](https://github.com/gruntwork-io/terragrunt/blob/master/util/file.go#L130) for checking if file.

File: options/options.go [hunk 2]
```diff
@@ -302,8 +302,40 @@ func (terragruntOptions *TerragruntOptions) Clone(terragruntConfigPath string) *
...
+	planFile := ""
+	var filteredArgs []string
+
+	for _, arg := range argsToInsert {
+		if checkIfPlanfile(arg) {
+			planFile = arg
+		} else {
+			filteredArgs = append(filteredArgs, arg)
+		}
+	}
```
Comment (yorinasub17) [inline options/options.go:327]: Wouldn't this also extract out var files?

**Review by yorinasub17 — COMMENTED** [review]
Thanks for your contribution! I had two comments regarding the implementation. Once those are addressed, I can kick off a build.

That said, this implementation (where you need to introspect if the arg is a plan file) makes me wonder if there should actually be a dedicated `plan_file` configuration in `terragrunt.hcl` or CLI (`--terragrunt-plan-file`).

File: options/options.go (hunk 1 above)
Comment (PertsevRoman) [reply options/options.go]: Hi @yorinasub17 ,
fixed

File: options/options.go (hunk 2 above)
Comment (PertsevRoman) [reply options/options.go:321]: @yorinasub17 , var files args are passed in `-var-file=filename` so `util.IsFile()` shouldn't return false

Comment (PertsevRoman) [reply options/options.go]: Also var files can't be specified with planfile

Comment (PertsevRoman) [reply options/options.go]: I've added tests which checks argument lists order

Comment (yorinasub17) [reply options/options.go]: You can pass in var files as `terragrunt -var-file filename` too, which I believe would fall into this case, right?

Comment (PertsevRoman) [reply options/options.go]: No, it'll fail with `flag provided but not defined: -var-file` error. Anyway, I'll special check for tfvars files

**Review by yorinasub17 — COMMENTED** [review]
Updates LGTM! Will kick off a build now.

**Review by yorinasub17 — APPROVED** [review]
Regression test passed, so will merge this in. Thanks again for your contribution!

**Comment by PertsevRoman** [comment]
@yorinasub17 thank you for help! Will be happy to join if there is some roadmap

**Pull request merged**